pkg/controller/login/reset_password.go

```diff
@@ -16,17 +16,60 @@
 package login
 
 import (
+	"context"
 	"net/http"
 
+	"github.com/google/exposure-notifications-verification-server/internal/firebase"
 	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller/flash"
 )
 
-func (c *Controller) HandleResetPassword() http.Handler {
+func (c *Controller) HandleShowResetPassword() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
+		c.renderResetPassword(ctx, w)
+	})
+}
+
+func (c *Controller) renderResetPassword(ctx context.Context, w http.ResponseWriter) {
+	m := controller.TemplateMapFromContext(ctx)
+	c.h.RenderHTML(w, "login/reset-password", m)
+}
 
+func (c *Controller) HandleSubmitResetPassword() http.Handler {
+	logger := c.logger.Named("login.HandleSubmitResetPassword")
+
+	type FormData struct {
+		Email string `form:"email"`
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		var form FormData
+		if err := controller.BindForm(w, r, &form); err != nil {
+			logger.Errorw("failed to bind form", "error", err)
+			controller.InternalError(w, r, c.h, err)
+			return
+		}
+
+		// There's no session yet, so make a one-time flash.
 		m := controller.TemplateMapFromContext(ctx)
-		m["firebase"] = c.config.Firebase
-		c.h.RenderHTML(w, "login/reset-password", m)
+		f := flash.New(nil)
+
+		if details, err := c.firebaseInternal.SendPasswordResetEmail(ctx, form.Email); err != nil {
+			// Treat not-found like success so we don't leak details.
+			if details.Error != firebase.EmailNotFound {
+				logger.Errorw("SendPasswordResetEmail failed", "error", err)
+				f.Error("reset password failed. %v", err)
+				c.renderResetPassword(ctx, w)
+				return
+			}
```

Comment:
Need to handle the case where the error is not `EmailNotFound`

Reply:
We do!

Reply:
Or rather we purposely skip over it and flash "Password reset email sent." so that they all look the same